chore(curriculum): add browsing the web effectively transcripts (#58223)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-browsing-the-web-effectively/672ac96491845f43ea9a26d7.md

Watch the lecture video and answer the questions below.

# --transcript--

What is the difference between a web browser, a website, and a search engine?

Web browsers are software applications that allow users to access and navigate the World Wide Web. They interpret and display web content, including text, images, videos, and interactive elements.

Key features of web browsers include the address bar, where you enter URLs or search queries, and the rendering engine, which translates HTML, CSS, and JavaScript into visual web pages.

> Browsers also offer bookmarks to save and organise favourite websites, extensions to enhance functionality, and various privacy and security features such as pop-up blockers and private browsing modes.

Reviewer: ```suggestion
Browsers also offer bookmarks to save and organize favorite websites, extensions to enhance functionality, and various privacy and security features such as pop-up blockers and private browsing modes.
```

Popular web browsers include Google Chrome, Mozilla Firefox, Microsoft Edge, Safari, and Opera.

Websites, on the other hand, are collections of web pages and related content that are identified by a common domain name and published on at least one web server. They are the destinations you visit using your web browser.

Components of websites include the domain name, which is the unique address of the website (like freecodecamp.org), and web pages, which are individual documents that make up a website.

Websites also often include multimedia content like images, videos, and audio files that enhance the user experience, as well as interactive elements such as forms and buttons that allow user interaction.

> Websites can be static, where content rarely changes, or dynamic, where content updates frequently or is personalised for users.

Reviewer: ```suggestion
Websites can be static, where content rarely changes, or dynamic, where content updates frequently or is personalized for users.
```

Search engines are web-based tools designed to help users find information on the internet. They use complex algorithms to index and rank web pages based on relevance to user queries.

Key aspects of search engines include web crawlers, also known as spiders or bots, which are programs that systematically browse the web to discover and index new content.

The indexing process involves organising and storing information about web pages in massive databases.

Search engines also employ ranking algorithms, which are complex systems that determine the order of search results based on factors like relevance, authority, and user experience. Popular search engines include Google, Bing, Yahoo, and DuckDuckGo.

The relationship between browsers, websites, and search engines is interconnected.

Web browsers are the tools used to access both websites and search engines.

Websites are the destinations that users visit through browsers or find via search engines.

> Search engines help users discover websites by providing organised access to the vast amount of information on the web.

Reviewer: ```suggestion
Search engines help users discover websites by providing organized access to the vast amount of information on the web.
```
